Google.Api.Gax.Grpc/GoogleCredentialExtensions.cs

```diff
@@ -0,0 +1,73 @@
+﻿/*
+ * Copyright 2024 Google LLC
+ * Use of this source code is governed by a BSD-style
+ * license that can be found in the LICENSE file or at
+ * https://developers.google.com/open-source/licenses/bsd
+ */
+
+using Google.Apis.Auth.OAuth2;
+using Grpc.Auth;
+using Grpc.Core;
+using System;
+using System.Threading;
+using System.Threading.Tasks;
+
+namespace Google.Api.Gax.Grpc
+{
+    /// <summary>
+    /// Extension methods for Google credential universe domain validation.
+    /// </summary>
+    public static class GoogleCredentialExtensions
```

Comment:
Do we definitely need this class to be public? Could we start with it internal, and then expose it if we find a need?

Reply:
(Or perhaps it could actually be an instance method in GoogleCredential in Google.Api.Auth?)